### ISSUE TITLE
Using C++ in VS-Code hassles & problems compared to Visual Studio 2019 community edition

### ISSUE THREAD

**Issue by RobFosterNYC** [microsoft/vscode#86525]
Ill get straight into it,
It is a complete hassle to be able to get C++ to work immediately & properly in VS-Code,
and having to install a bunch of things like G++ GCC and mingw.

Getting this message:

> **#include errors detected. Please update your includePath.** 

With a bunch of squiggly lines of errors.
Why is it such a pain and hassles to get C++ working properly without having to jump through so many hoops to be able to just going with C++ and code without having to install tons of things first.
And for some reason VS-Code cant compile C++ without having to go through so much crap
installing so many things to get it to work.

Can Microsoft please figure this out and fix this so people dont have to go through this.?
I would like to just install the C++ plugin in VS-Code and start coding & compiling without having to go through hours with this hassle.

The sad part is, When I install Visual Studio 2019 community edition,
I can easily use C++ without all the hassles and heaps of problems.  But when using VS-Code it becomes a very huge problem & hassle to get C++ to work in VS_Code.  Why ?
If Visual Studio 2019 can do it, why cant VS-Code cant ?
I would like Microsoft people to look into this and hopefully fix this small problem,
So you guys can make millions of people happy, again to be able to use C++ in VS-Code just like
Visual Studio 2019 community edition.

**Comment by DeltaRazero** [comment]
Hi I happened to see your question,

First of all, I assume you use https://github.com/microsoft/vscode-cpptools , so I suggest you open an issue at that repo instead of the vscode repo. This VSCode repo is purely meant for the application itself and its main components (like the editor) and embedded extensions. You'll get a better response at the cpptools repo.

Second of all:
> If Visual Studio 2019 can do it, why cant VS-Code cant ?

Because Visual Studio is an IDE and Visual Studio Code (at its core) is 'just' an editor. I suggest reading [this answer on Atom's site](https://discuss.atom.io/t/what-is-the-difference-between-an-ide-and-an-editor/32629).
Visual Studio and Visual Studio Code's goals as application are fundamentally different.

Third, if you have trouble configuring and installing the necessary tools I suggest you read [this article if you want to use mingw](https://code.visualstudio.com/docs/cpp/config-mingw) or [this article if you want to use msvc](https://code.visualstudio.com/docs/cpp/config-msvc); there is enough [documentation to help you get started](https://code.visualstudio.com/docs/languages/cpp).

It might be possible to create a command in Microsoft's C++ extension to download and configure mingw or clang (msvc is perhaps out of the question due to the distribution system) and let it configure some things by itself, but in my opinion I don't think that is necessary.

I hope this helps you setting up your C++ environment for in use with VSCode and understand the difference between VS and VSCode.

**Comment by SirTwsted** [comment]
I am also having the same problem, as with thousands of other people.
Wow, very condescending of you to respond and talk down to people like that.
You might know more than the other person, it doesn't give you the right to be condescending to them.
Not to mention that every other extension has a 1 simple installation solution,
and people don't have to spend an hour or more reading through large text documents just to configure 1 thing they want to work in visual studio code.
This extension tends to be more of a chore than anything else to get it to work.
And people shouldn't have to go through all that, just to get an extension to work.

I found its much much less of a hassle to just use visual studio 2019,
until Microsoft figures out a way that prevents people from having to spend such an enormous amount of time just to get an extension to work & configured properly.
I have talked with many very very experienced people in C++ and even they believe that the current manner of installing and configuring C++ in visual studio code is a pain & annoyance compared 
to many other extensions available for visual studio code and the way they install compared to C++ extension.

In this day in age almost going on 2020, it shouldn't be this painful to install an extension and get it to work properly in visual studio code, without going through so many hurtles and read large text file documents to configure an extension in visual studio code.
I brought this situation to the attention of a few very experienced software engineers to ask them for help hand,
Those experienced people also agree with me on this, that this is a hassle in this day in age.
and they work for very large companies and have over 25 years experience.

anyways, that's my take and 2-cents on the matter.
Hopefully Microsoft developers look into this and finds a way to alleviate this small problem,
that's giving many people a migraine. Because it shouldn't be this annoying.
